benchexec/resources.py

```diff
@@ -84,13 +84,26 @@ def get_cpu_cores_per_run(coreLimit, num_of_threads, use_hyperthreading, my_cgro
 
         logging.debug("List of available CPU cores is %s.", allCpus)
 
+        # read mapping of core to memory region
+        memory_regions = []
+        for core in allCpus:
+            coreDir = '/sys/devices/system/cpu/cpu{0}/'.format(core)
+            memory_regions.append(_get_memory_banks_listed_in_dir(coreDir)[0])
+        cores_of_memory_region = collections.defaultdict(list)
+        for core, memory_region in zip(allCpus, memory_regions):
+            cores_of_memory_region[memory_region].append(core)
+        logging.debug("Memory regions of cores are %s.", cores_of_memory_region)
+
         # read mapping of core to CPU ("physical package")
         physical_packages = [get_cpu_package_for_core(core) for core in allCpus]
         cores_of_package = collections.defaultdict(list)
         for core, package in zip(allCpus, physical_packages):
             cores_of_package[package].append(core)
         logging.debug("Physical packages of cores are %s.", cores_of_package)
 
+        # select the more fine grained division among memory regions and physical package
+        cores_of_unit = (cores_of_memory_region if (len(cores_of_memory_region) >= len(cores_of_package)) else cores_of_package)
```

Comment:
Could you please add a debug log statement here, if `cores_of_memory_region` is used?

I guess rewriting into an `if ...: ... else:` construct is then more clear.